### PR TITLE
Implement socket.io into application and add functionality to button …

### PR DESCRIPTION
**Pull request by Pma913**
…for making food truck live

____ Wrote Tests ___x_ Implemented ____ Reviewed

## Neccesary checkmarks
- [ ] All Tests are Passing in all environments
- [x] The code will run locally
## Type of change
- [x] New feature
- [ ] Bug Fix
- [ ] Refactor
## Description of Change:

  description: This will add sockets.io to the project. Sets up the connection to the express server and server to the app. Creates handshake between two or more devices running. Enables vendor view to be able to toggle live or closed and transmits data real time to other devices and updates state with new data. There is a bug that causes the app to crash when the window is refreshed in the vendor view. 
    
## Requested feedback: 
    
  I'd like feedback on... anywhere that could use it. Especially around the websocket

## Check the correct boxes

- [ ] This broke nothing
- [x] This broke some stuff
- [ ] This broke everything

## Testing Changes

- [x] No Tests have been changed
- [ ] Some Tests have been changed
- [ ] All of the Tests have been changed(Please describe what in the world happened)

## Checklist:

- [ ] My code has no unused/commented out code
- [ ] My code has no binding.pry's
- [ ] I have reviewed my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have fully tested my code

(For Fun!)Please Include a link to a gif of your feelings about this branch

Link:

### CODE REVIEW

File: src/Components/App/App.js [hunk 1]
```diff
@@ -17,6 +18,23 @@ const App = () => {
...
+      }
+  })
+
+  const sendMessage = (truck) => {
+    if (truck.status === 'false') {
+      truck.status = 'true'
+    } else {
+      truck.status = 'false'
+    }
+      socket.emit("send_data", { updatedVendors: vendors.filter(v => v.id !== truck.id), truck: truck });
```
Comment (Devynnem) [inline src/Components/App/App.js:35]: This is really cool, thanks for working through this!

File: src/Components/App/App.js [hunk 2]
```diff
@@ -70,20 +89,30 @@ const App = () => {
...
+    setVendors(thisData)
+    // this is going to fetch the data and then set state but then also reset isLoading to false
+  }
+
   useEffect(() => {
-    fetchData();
-// call in mock data somehow and establish it as the state of vendors
+    fetchData()
+    
+    socket.on('receive_data', (data) => {
```
Comment (Devynnem) [inline src/Components/App/App.js:109]: It's interesting/makes sense that is also has to receive the message or data before it shows it on the DOM

File: src/Components/App/App.js [hunk 3]
```diff
@@ -70,20 +89,30 @@ const App = () => {
...
+      setVendors([data.truck, ...data.updatedVendors])
+    });
+
+    return () => {
+      socket.off('receive_data');
```
Comment (Devynnem) [inline src/Components/App/App.js:114]: I still need to look up and understand more what socket.off

**Review by Devynnem — APPROVED** [review]

**Comment by JacobMacFarlane** [comment]
Awesome work getting this implemented patrick!

**Pull request merged**